### ISSUE TITLE
package resource not uninstalling MSI without `source` property defined

### ISSUE THREAD

**Issue by jmery** [chef/chef#5274]
## Description

This works properly and uninstalls the package:
```
package 'ActivePerl Enterprise 5.16.3 Build 1603.4 (64-bit)' do
  action :remove
  installer_type :msi
  source 'https://s3.amazonaws.com/jmery/pepsi-poc/ActivePearl.msi'
  checksum 'e6aca518673e37606990a3db368d401dc860182a3da09a9d77770ba16f24dfe9'
  options '/q'
end
```

This does NOT work.  The client converges and says it's uninstalling, but no action taken.
```
package 'ActivePerl Enterprise 5.16.3 Build 1603.4 (64-bit)' do
  action :remove
  installer_type :msi
#  source 'https://s3.amazonaws.com/jmery/pepsi-poc/ActivePearl.msi'
#  checksum 'e6aca518673e37606990a3db368d401dc860182a3da09a9d77770ba16f24dfe9'
  options '/q'
end
```

It appears that either the `package` resource or the MSI itself are not performing a `/quiet` uninstall.  This was verified with other MSI packages from the command line (e.g. grab the MSI `UninstallString` from the registry and run from command prompt).  Tested with the chef-client install as well with the same results.

## Chef Version

chef-client version  `Chef: 12.13.37`

## Platform Version

Windows 2012 Standard x64 (note:  **Not** 2012 R2)

## Replication Case

Install the package above in the manner of your choice (I used the `package` resource to install).  Run the above `package` resource code to attempt to uninstall.

The package `source` above is wide open for use and the `checksum` is valid.

## Client Output

No errors reported

## Stacktrace

N/A

CC @mwrock

**Comment by Shafiud** [comment]
i don't see any difference in the code between "This works properly" Vs "This does NOT work".

**Comment by jmery** [comment]
`source` and `checksum` are commented on the non-working resource.  Essentially, `source` must be specified for `action :remove` to work.

**Comment by thommay** [comment]
@chef/client-windows please take a look…

**Comment by Aliasgar16** [comment]
@jmery I successfully reproduced this issue for package `ActivePerl Enterprise 5.16.3 Build 1603.4 (64-bit)`. I found that `chef` was correctly handling the scenario (when removing the package with no source value passed) by using the `uninstallString` retrieved from the `Windows Registry`, but the `uninstallString` for the package `ActivePerl Enterprise 5.16.3 Build 1603.4 (64-bit)` itself is invalid as it contains `/I` option (which is used for installation) instead of `/X` option (which is used for un-installation). I will modify the `chef` code to handle such scenarios and some other similar kind of scenarios (if I found any). Will update you further.
